erpnext/stock/doctype/stock_ledger_entry/stock_ledger_entry.py

```diff
@@ -170,6 +170,42 @@ def validate_with_last_transaction_posting_time(self):
 					msg += "<br>" + "<br>".join(authorized_users)
 					frappe.throw(msg, BackDatedStockTransaction, title=_("Backdated Stock Entry"))
 
+	def validate_future_sle_by_batch_for_outgoing_entries(self):
+		if not self.batch_no or self.actual_qty is None or self.actual_qty >= 0:
+			return
+
+		if self.get("allow_negative_stock"):
+			return
+
+		voucher_condition = ""
+		voucher_no = self.get("voucher_no")
+		voucher_type = self.get("voucher_type")
+		if voucher_no and voucher_type:
+			voucher_condition = f"and not (voucher_no = '{voucher_no}' and voucher_type = '{voucher_type}')"
+
+		future_negative_entries = frappe.db.sql("""select
+					voucher_type, voucher_no, item_code, warehouse, posting_date
+				from `tabStock Ledger Entry`
+				where
+					warehouse = %(warehouse)s
+					and item_code = %(item_code)s
+					and batch_no = %(batch_no)s
+					and (qty_after_transaction + %(actual_qty)s) < 0
```

Comment:
`qty_after_transaction` represents total quantity, so this isn't correct for multi-batch case.


`actual_amt_check` check on stock ledger entry

Reply:
IMO let's just get rid of this whole function and expand batch-wise check to consider running totals instead of final sum.